modules/ngx_http_upstream_session_sticky_module/ngx_http_upstream_session_sticky_module.c

```diff
@@ -246,16 +234,8 @@ ngx_http_upstream_session_sticky_init_peer(ngx_http_request_t *r,
     sspd->sscf = sscf;
     sspd->get_rr_peer = ngx_http_upstream_get_round_robin_peer;
 
-    r->upstream->peer.data = sspd;
     r->upstream->peer.get = ngx_http_upstream_session_sticky_get_peer;
 
-#if (NGX_HTTP_SSL)
-    r->upstream->peer.set_session =
-                            ngx_http_upstream_session_sticky_set_peer_session;
-    r->upstream->peer.save_session =
-                            ngx_http_upstream_session_sticky_save_peer_session;
-#endif
```

Comment:
Do not need to re-write set_session() and save_session() function when u->peer.data is point of ngx_http_upstream_rr_peer_data_t, see ngx_http_upstream_ip_hash_module.c.

Only when u->peer.data is changed to module private data, it needs to re-write these two callbacks. For example, see ngx_http_upstream_keepalive_module.c.